### PR TITLE
Unify the syntax format for CSS color functions

### LINKED ISSUE #26434: Inconsistency in the format among CSS color functions
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/hsl

### What specific section or headline is this issue about?

Syntax

### What information was incorrect, unhelpful, or incomplete?

Currently there're rather different formats for the "Syntax" section of CSS color functions.

For the contents immediately below the "Syntax" heading and above the "Values" heading:

  1. [`color()`](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/color), [`color-contrast()`](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/color-contrast) (the name even deviates from [that](https://w3c.github.io/csswg-drafts/css-color-6/#colorcontrast) in the premature spec), [`color-mix()`](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/color-mix), [`hwb()`](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/hwb), [`lab()`](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/lab), [`lch()`](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/lch), [`oklab()`](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/oklab), and [`oklch()`](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/oklch) demonstrate the syntax with concrete values only.
  2. [`hsl()`](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/hsl) demonstrates various possible syntaxes with explaining comments *but* without concrete values in the code block, followed by an apparently redundent explanation on the syntax.
  3. [`rgb()`](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/rgb) demonstrates various possible syntaxes with explaining comments *and* concrete values in the code block, followed by a explanation on the syntax.

`hsl()` is almost surely not doing the right thing here as this section usually starts with concrete examples, but I'm not sure which of the rest is better due to what comes in the "Values" subsection:

  1. `color()`, `color-contrast()`, `hwb()` start with the line "Functional notation: `notation`", followed by a definition list explaining the components in `notation`.
  2. `color-mix()`, `lab()`, `lch()`, `oklab()`, `oklch()` do the same, but with an apparently failed attempt to use definition lists.
  3. `hsl()` and `rgb()` explain the components already named above this subsection.

I'd like to open a PR to unify the format since it blocks the l10n of these pages, but I have to get informed on the preferred one.

### What did you expect to see?

A consistent format for this section across color functions.

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### PR DESCRIPTION
**Pull request by yarusome**
### Description

This PR unifies the format for the "Syntax" sections of CSS color functions.

### Motivation

See #26434.

### Related issues and pull requests

Fixes #26434.

### CODE REVIEW

File: files/en-us/web/css/color_value/color-mix/index.md [hunk 1]
```diff
@@ -20,23 +20,25 @@ color-mix(in hsl longer hue, hsl(120 100% 50%) 20%, white);
...
-  - `<color>` is any valid {{cssxref("color_value","color")}}
+- `color`
 
-  - `<percentage>` is a number between 0 and 100 with an optional `%` sign.
-    If no percentage is specified, the default is 50%.
+  - : Any valid {{CSSXref("&lt;color&gt;")}}.
+
+- `percentage` {{optional_inline}}
+
+  - : A {{CSSXref("&lt;percentage&gt;")}} between `0%` and `100%`. See [percentage normalization](https://w3c.github.io/csswg-drafts/css-color-5/#color-mix-percent-norm) for various cases of effects for specifying these two percentages.
```
Comment (yarusome) [inline files/en-us/web/css/color_value/color-mix/index.md:41]: Should the details be copied from the spec here?

File: files/en-us/web/css/color_value/color-contrast/index.md [hunk 2]
```diff
@@ -20,11 +20,15 @@ color-contrast(#008080 vs olive, var(--myColor), #d2691e)
 
 ### Values
 
-- Functional notation: `color-contrast( <color> vs <color>#{2,} )`
+Functional notation: `color-contrast(color vs color-list)`
```
Comment (teoli2003) [inline files/en-us/web/css/color_value/color-contrast/index.md:23]: One problem with dropping the `<>` is that now it is difficult to know what has to be replaced (`color` and `color-list`) and what has to be typed as is (`vs`)

File: files/en-us/web/css/color_value/color-mix/index.md (hunk 1 above)
Comment (teoli2003) [reply files/en-us/web/css/color_value/color-mix/index.md:41]: Not copied but explained in layman's language (The spec speech is far more difficult to understand for web developers in this case as it describes the algorithm steps and not the outcome). I want to avoid links to the spec inside the text, especially to versioned links, as these get outdated over time.

**Review by teoli2003 — DISMISSED** [review]

File: files/en-us/web/css/color_value/color-contrast/index.md [hunk 3]
```diff
@@ -20,11 +20,15 @@ color-contrast(#008080 vs olive, var(--myColor), #d2691e)
...
 
-  - : `<color>` is any valid {{cssxref("color_value","color")}}.
+- `color`
 
-    `<color>#{2,}` is a comma-separated list of color values to compare with the first value.
+  - : Any valid {{CSSXref("&lt;color&gt;")}}.
```
Comment (yarusome) [inline files/en-us/web/css/color_value/color-contrast/index.md:27]: @teoli2003 How about this ("literal keyword/string/token"?):
```suggestion
  - : Any valid {{CSSXref("&lt;color&gt;")}}.

- `vs`

  - : A literal token as a component of the syntax.
```

**Comment by yarusome** [comment]
@bsmth I added literal tokens in the "Values" section after teoli2003's comment. Is it appropriate now? Or the functions with literal tokens can use the CSS syntax style instead.

P.S. The missing component issue raised in #26434 might be deferred to future PRs.

**Comment by bsmth** [comment]
> I added literal tokens in the "Values" section after teoli2003's comment. Is it appropriate now?

Taking a look now 👀

File: files/en-us/web/css/color_value/color-contrast/index.md (hunk 2 above)
Comment (bsmth) [reply files/en-us/web/css/color_value/color-contrast/index.md:23]: We're already omitting the angle brackets on some of the existing function pages

https://github.com/mdn/content/blob/main/files/en-us/web/css/color_value/hwb/index.md?plain=1#L25

instead of `Functional notation: `hwb(<hue> <whiteness> <blackness>[ / <alpha>])`

I think the legibility is good as-is, happy to hear your opinion!

File: files/en-us/web/css/color_value/color-mix/index.md [hunk 4]
```diff
@@ -20,23 +20,39 @@ color-mix(in hsl longer hue, hsl(120 100% 50%) 20%, white);
...
 
-    - `<rectangular-color-space>` or
-    - `<polar-color-space>` and an optional `<hue-interpolation-method>`.
+  - : A literal token as a component of the syntax.
 
-  - `<rectangular-color-space>` is one of `srgb`, `srgb-linear`, `lab`, `oklab`, `xyz`, `xyz-d50`, `xyz-d65`.
+- `colorspace`
 
-  - `<polar-color-space>` is one of `hsl`, `hwb`, `lch`, `oklch`.
+  - : One of `srgb`, `srgb-linear`, `lab`, `oklab`, `xyz`, `xyz-d50`, `xyz-d65`, `hsl`, `hwb`, `lch`, and `oklch`, specifying the color space for interpolation.
```
Comment (bsmth) [inline files/en-us/web/css/color_value/color-mix/index.md:31]:
```suggestion
  - : One of `srgb`, `srgb-linear`, `lab`, `oklab`, `xyz`, `xyz-d50`, `xyz-d65`, `hsl`, `hwb`, `lch`, or `oklch`, specifying the color space for interpolation.
```

File: files/en-us/web/css/color_value/color/index.md [hunk 5]
```diff
@@ -22,13 +22,19 @@ color(display-p3 1 0.5 0 / .5);
...
 ### Values
 
-- Functional notation: `color( [ [<ident> | <dashed-ident>]? [ <number-percentage>+ ] [ / <alpha-value> ]? ] )`
+Functional notation: `color([colorspace] params[ / A])`
 
-  - : `[<ident> | <dashed-ident>]` is an optional {{cssxref("ident")}} or {{cssxref("dashed-ident")}} denoting the colorspace. If this is an `<ident>` it denotes one of the predefined colorspaces (such as display-p3); if it is a `<dashed-ident>` it denotes a custom colorspace, defined by a [`@color-profile`](/en-US/docs/Web/CSS/@color-profile) rule.
+- `colorspace` {{optional_inline}}
 
-    `[ <number-percentage>+ ]` is one or more {{cssxref("number")}} or {{cssxref("percentage")}} values providing the parameter values that the colorspace takes.
+  - : An {{CSSXref("&lt;ident&gt;")}} or {{CSSXref("&lt;dashed-ident&gt;")}} denoting the color space. If this is an `<ident>` it denotes one of the predefined color spaces (such as display-p3); if it is a `<dashed-ident>` it denotes a custom color space, defined by a {{CSSXref("@color-profile")}} rule.
```
Comment (bsmth) [inline files/en-us/web/css/color_value/color/index.md:29]: I guess we have a finite list of predefined colorspace values srgb, srgb-linear, display-p3, a98-rgb, prophoto-rgb, rec2020. I feel like we should mention these

File: files/en-us/web/css/color_value/hwb/index.md [hunk 6]
```diff
@@ -20,17 +20,23 @@ hwb(194 0% 0% / .5) /* #00c3ff with 50% opacity */
...
+  - : An {{cssxref("&lt;angle&gt;")}} of the color circle given in `deg`s, `rad`s, `grad`s, or `turn`s in the [CSS Color](https://drafts.csswg.org/css-color/#typedef-hue) specification. When written as a unitless {{cssxref("&lt;number&gt;")}}, it is interpreted as degrees, as specified in the [CSS Color Level 3](https://drafts.csswg.org/css-color-3/#hsl-color) specification. By definition, red=0deg=360deg, with the other colors spread around the circle, so green=120deg, blue=240deg, etc. As an `<angle>`, it implicitly wraps around such that -120deg=240deg, 480deg=120deg, -1turn=1turn, etc.
```
Comment (bsmth) [inline files/en-us/web/css/color_value/hwb/index.md:27]: This is another section where we're defining hue again, this could be redundant.

File: files/en-us/web/css/color_value/color/index.md (hunk 5 above)
Comment (yarusome) [reply files/en-us/web/css/color_value/color/index.md:29]: The [spec](https://w3c.github.io/csswg-drafts/css-color-5/#color-function) also lists XYZ color spaces (`xyz`, `xyz-d50`, `xyz-d65`) along these RGB color spaces. Should they be added here, too?

File: files/en-us/web/css/color_value/hwb/index.md (hunk 6 above)
Comment (yarusome) [reply files/en-us/web/css/color_value/hwb/index.md:27]: For problems existing in most CSS color pages (like the "missing component" one), they could be tackled in following PRs as this PR is already trying to solve such a widespread problem :)

File: files/en-us/web/css/color_value/hwb/index.md [hunk 7]
```diff
@@ -20,17 +20,23 @@ hwb(194 0% 0% / .5) /* #00c3ff with 50% opacity */
...
+  - : A {{CSSXref("&lt;percentage&gt;")}} representing whiteness that specifies the amount of white to mix in, from 0% (no whiteness) to 100% (full whiteness).
 
-    `A` (alpha) can be a {{cssxref("&lt;number&gt;")}} between `0` and `1`, or a {{cssxref("&lt;percentage&gt;")}}, where the number `1` corresponds to `100%` (full opacity).
+- `B`
+
+  - : A {{CSSXref("&lt;percentage&gt;")}} representing blackness that specifies the amount of black to mix in, also from 0% (no blackness) to 100% (full blackness).
+
+- `A` {{optional_inline}}
+
+  - : An {{CSSXref("&lt;alpha-value&gt;")}}, where the number `1` corresponds to `100%` (full opacity).
```
[the remark this answers is not in the document]
Comment (yarusome) [reply files/en-us/web/css/color_value/hwb/index.md:37]: I added descriptions for the effective normalization. Do I have to add one of the palettes from Section 8.3 (i.e. the snapshot here)?

File: files/en-us/web/css/color_value/color/index.md (hunk 5 above)
Comment (bsmth) [reply files/en-us/web/css/color_value/color/index.md:29]: Indeed, good catch 👍🏻

File: files/en-us/web/css/color_value/hwb/index.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (bsmth) [reply files/en-us/web/css/color_value/hwb/index.md:37]: No we don't need to, this was just for illustration.

>I added descriptions for the effective normalization.

Excellent, looks great

**Review by bsmth — APPROVED** [review]
I'm approving now pending <strike>two comments</strike> https://github.com/mdn/content/pull/26448#discussion_r1179944663, happy to merge when resolved, well done on this cleanup! 🙏🏻

**Comment by yarusome** [comment]
@bsmth I found a few problems regarding `color()`:

  1. Its spec link in BCD points to CSS Colors Level 4, but in Level 4 this functions doesn't accept custom color spaces.
  2. In Level 5, it accepts not only custom color spaces but also [relative colors](https://w3c.github.io/csswg-drafts/css-color-5/#color-function), i.e. the `[from <color>]?` component.

Which level does the "Syntax" section try to describe?

**Comment by bsmth** [comment]
Well spotted, I did not notice this: 

>Removed left-over mention of custom color spaces [(feature was moved to CSS Color 5)](https://w3c.github.io/csswg-drafts/css-color/#changes-from-20220705) 

We should remove mention of this as we're targeting level 4 functionality at the moment

**Comment by bsmth** [comment]
Thanks @yarusome - I'm going to merge and bring up the syntax discussion separately as a follow-up as I think it's worth getting these other improvements in ASAP. Well done 👍🏻
